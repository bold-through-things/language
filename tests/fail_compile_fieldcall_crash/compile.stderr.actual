[
  {
    "recoverable": false,
    "line": 24,
    "char": 0,
    "content": "67lang:call heartbeat",
    "error": "failed to assert: could not find a matching overload for heartbeat with arguments []",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": []
  },
  {
    "recoverable": false,
    "line": 15,
    "char": 0,
    "content": "67lang:call event",
    "error": "failed to assert: could not find a matching overload for event with arguments []",
    "error_type": "NO_MATCHING_OVERLOAD",
    "visible_overloads": [
      {
        "convention": {
          "convention": "FieldCall",
          "field": "event",
          "demands": [
            {
              "name": "HTMLScriptElement",
              "type_params": [],
              "fields": []
            }
          ],
          "returns": {
            "name": "str",
            "type_params": [],
            "fields": []
          }
        },
        "reasons": [
          {
            "reason_type": "ARITY_MISMATCH",
            "demanded_arity": 1,
            "received_arity": 0
          }
        ]
      },
      {
        "convention": {
          "convention": "FieldCall",
          "field": "event",
          "demands": [
            {
              "name": "HTMLScriptElement",
              "type_params": [],
              "fields": []
            },
            {
              "name": "str",
              "type_params": [],
              "fields": []
            }
          ],
          "returns": {
            "name": "None",
            "type_params": [],
            "fields": []
          }
        },
        "reasons": [
          {
            "reason_type": "ARITY_MISMATCH",
            "demanded_arity": 2,
            "received_arity": 0
          }
        ]
      },
      {
        "convention": {
          "convention": "FieldCall",
          "field": "event",
          "demands": [
            {
              "name": "Window",
              "type_params": [],
              "fields": []
            }
          ],
          "returns": {
            "name": "Event",
            "type_params": [],
            "fields": []
          }
        },
        "reasons": [
          {
            "reason_type": "ARITY_MISMATCH",
            "demanded_arity": 1,
            "received_arity": 0
          }
        ]
      }
    ]
  }
]
