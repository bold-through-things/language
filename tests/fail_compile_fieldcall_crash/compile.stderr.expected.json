{
  "$items": [
    {
      "recoverable": false,
      "line": 24,
      "char": 0,
      "content": { "$contains": "call heartbeat" },
      "error_type": "NO_MATCHING_OVERLOAD",
      "visible_overloads": { "$len": 0 }
    },
    {
      "recoverable": false,
      "line": 15,
      "char": 0,
      "content": { "$contains": "call event" },
      "error_type": "NO_MATCHING_OVERLOAD",
      "visible_overloads": { "$len": 3 }
    }
  ],
  "$order": "any"
}
